Makefile

```diff
@@ -478,23 +473,6 @@ opt.opt:
 	$(MAKE) otherlibrariesopt
 	$(MAKE) ocamllex.opt ocamltoolsopt ocamltoolsopt.opt $(OCAMLDOC_OPT) \
 	  ocamltest.opt
-else
-opt.opt: core opt-core ocamlc.opt all ocamlopt.opt ocamllex.opt \
-         ocamltoolsopt ocamltoolsopt.opt otherlibrariesopt $(OCAMLDOC_OPT) \
-         ocamltest.opt
-endif
```

Comment:
Maybe this could be considered as not for this GPR, but if `opt.opt` is being tidied up then perhaps it should go the whole hog. There's no earthly reason why the target for "building the native-code versions of the tools" should run `checkstack`, `runtime`, `core`, `ocaml`, `opt-core`, `otherlibraries $(WITH_DEBUGGER) $(WITH_OCAMLDOC) ocamltest`, `otherlibrariesopt` or `ocamltoolsopt`.

I think most of this is here because `world.opt` is lazy? I think `tests` and `manual-pregen` should probably depend on `world.opt` rather than `opt.opt` if it is changed.

Reply:
At some point we'll probably decide between making each of these targets self-sufficient, or properly documenting their dependencies. And then I hope we'll eliminate most of them.
But this is for another PR.